README.md

```diff
@@ -113,17 +114,37 @@ julia> residual!(model, model.meta.x0, model.F)
  -0.4486499211288866
 ```
 
-You need to call ``jac_structure_residual!`` at least once before calling ``jac_op_residual!``.
+```julia
+julia> meta_nls = nls_meta(model)
+julia> S = typeof(model.meta.x0)
+julia> F = S(undef, meta_nls.nequ)
+julia> residual!(model, model.meta.x0, F)
+63686-element Vector{Float64}:
+ -9.020226301243213
+ 11.263958304987227
+  ⋮
+ -0.01443314653496941
+ -0.4486499211288866
+```
+
+You need to call `jac_structure_residual!` at least once before calling `jac_op_residual`.
```

Comment:
why?

Reply:
The `jac_op_residual` I use is based on the sparse matrix so I need to create it before using it because it doens't do it himself.

Reply:
But I'm confused because when you call `jac_op_residual` below (on line 157), you don't use `rows`, `cols` or `vals`.

Reply:
You are right, and I didn't even test if the last `jac_op_residual` was wroking correctly. I will digger into this to see if it works.